### PR TITLE
KEYCLOAK-6827 Performance test for user registration

### PR DESCRIPTION
**Pull request by tkyjovsk**

### CODE REVIEW

File: testsuite/performance/tests/src/test/scala/keycloak/OIDCLoginAndLogoutSimulation.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import io.gatling.http.Predef._
+import keycloak.OIDCScenarioBuilder._
+
+import org.keycloak.performance.TestConfig
+
+
+/**
+  * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+  * @author Marko Strukelj &lt;mstrukel@redhat.com&gt;
+  */
```
Comment (mstruk) [inline testsuite/performance/tests/src/test/scala/keycloak/OIDCLoginAndLogoutSimulation.scala]: Let's throw these @author tags out. They lost all relationship with original code already :)

File: testsuite/performance/tests/src/test/scala/keycloak/OIDCRegisterAndLogoutSimulation.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import io.gatling.http.Predef._
+import keycloak.OIDCScenarioBuilder._
+
+import org.keycloak.performance.TestConfig
+
+
+/**
+  * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+  * @author Marko Strukelj &lt;mstrukel@redhat.com&gt;
+  */
```
Comment (mstruk) [inline testsuite/performance/tests/src/test/scala/keycloak/OIDCRegisterAndLogoutSimulation.scala]: Let's throw these @author tags out.

**Review by mstruk — CHANGES_REQUESTED** [review]

File: testsuite/performance/tests/src/main/java/org/keycloak/performance/TestConfig.java [hunk 3]
```diff
@@ -163,7 +191,12 @@ public synchronized UserInfo next() {
 
                 String user = computeUsername(realm, idx);
                 idx += 1;
-                return new UserInfo(user, computePassword(user));
+                return new UserInfo(user, 
+                        computePassword(user),
+                        computeFirstName(idx),
```
Comment (mstruk) [inline testsuite/performance/tests/src/main/java/org/keycloak/performance/TestConfig.java]: Firstname here uses idx incremented by 1 already and is thus out of sync with username, and email. 
One fix is to pass idx-1. Another one is to move 'idx += 1' line into a finally block around return statement.

Comment (tkyjovsk) [reply testsuite/performance/tests/src/main/java/org/keycloak/performance/TestConfig.java]: Good catch.

File: testsuite/performance/tests/src/test/scala/keycloak/OIDCRegisterAndLogoutSimulation.scala [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package keycloak
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import keycloak.OIDCScenarioBuilder._
+
+import org.keycloak.performance.TestConfig
+
+
```
Comment (tkyjovsk) [inline testsuite/performance/tests/src/test/scala/keycloak/OIDCRegisterAndLogoutSimulation.scala]: @author tags removed

File: testsuite/performance/README.md [hunk 5]
```diff
@@ -206,12 +213,30 @@ When running the tests it is also necessary to define a dataset to use. Usage is
...
 - Run test specific test and dataset parameters:
 
-`mvn verify -P test -Dtest.properties=basic-oidc -Ddataset=100u2c`
+`mvn verify -P test -Dtest.properties=oidc-login-logout -Ddataset=100u2c`
 
 - Run test with specific test and dataset parameters, overriding some from command line:
 
 `mvn verify -P test -Dtest.properties=admin-console -Ddataset=100u2c -DrampUpPeriod=30 -DwarmUpPeriod=60 -DusersPerSec=0.3`
 
+#### Running `OIDCRegisterAndLogoutSimulation`
```
Comment (tkyjovsk) [inline testsuite/performance/README.md]: Added a note on how to run registration test.

File: testsuite/performance/tests/src/main/java/org/keycloak/performance/TestConfig.java [hunk 6]
```diff
@@ -162,8 +190,14 @@ public synchronized UserInfo next() {
                 }
 
                 String user = computeUsername(realm, idx);
+                String firstName= computeFirstName(idx);
                 idx += 1;
```
Comment (tkyjovsk) [inline testsuite/performance/tests/src/main/java/org/keycloak/performance/TestConfig.java]: username / firstName should now be consistent

**Review by mstruk — APPROVED** [review]

**Pull request merged**